lib/Arch/X86/Arch.cpp

```diff
@@ -788,36 +788,26 @@ static void DecodeOperand(Instruction &inst, const xed_decoded_inst_t *xedd,
   }
 }
 
-static uint64_t BytesToCallBits(const uint8_t *bytes) {
-  // The pattern we're looking for is 5 bytes long.
-  uint64_t bits = 0;
-  bits = (bits << 8) | static_cast<uint32_t>(bytes[0]);
-  bits = (bits << 8) | static_cast<uint32_t>(bytes[1]);
-  bits = (bits << 8) | static_cast<uint32_t>(bytes[2]);
-  bits = (bits << 8) | static_cast<uint32_t>(bytes[3]);
-  bits = (bits << 8) | static_cast<uint32_t>(bytes[4]);
-  return bits;
-}
+static unsigned char GET_PC_IDIOM_CALL[] = {0xE8, 0x00, 0x00, 0x00, 0x00};
 
 static bool TryDecodeGetPCIdiom(Instruction &inst) {
   // Check for CALL+POP idiom used to retrieve PC.
   if (inst.bytes.size() != 6) {
```

Comment:
This feels overly restrictive. I think it should be saying: if we don't have /at least/ six bytes, then we can't find our idiom. If we have at least 6, then we can. The follow-on issue is that if we have more than 6 bytes, then we might have a different form of POP on our hands. It seems reasonable-ish to me to generically handle this, e.g. have a static const mapping of ISEL names of supported trailing pops, mapping to isel names for the new idiom.